config/config_test.go

```diff
@@ -87,23 +106,43 @@ func TestValidate(t *testing.T) {
 		}
 		cfg, err := newConfig(cfgService)
 		test.That(t, err, test.ShouldBeNil)
-		test.That(t, cfg.DataDirectory, test.ShouldEqual, cfgService.Attributes["data_dir"])
 		test.That(t, cfg.Sensors, test.ShouldResemble, cfgService.Attributes["sensors"])
 		test.That(t, cfg.DataRateMsec, test.ShouldEqual, cfgService.Attributes["data_rate_msec"])
-		test.That(t, *cfg.MapRateSec, test.ShouldEqual, cfgService.Attributes["map_rate_sec"])
 		test.That(t, cfg.ConfigParams, test.ShouldResemble, cfgService.Attributes["config_params"])
+
+		if cloudStoryEnabled {
+			test.That(t, cfg.ExistingMap, test.ShouldEqual, cfgService.Attributes["existing_map"])
+			test.That(t, cfg.MapRateSec, test.ShouldBeNil)
+		} else {
+			test.That(t, cfg.DataDirectory, test.ShouldEqual, cfgService.Attributes["data_dir"])
+			test.That(t, *cfg.MapRateSec, test.ShouldEqual, cfgService.Attributes["map_rate_sec"])
+		}
 	})
 }
 
+func TestValidate(t *testing.T) {
+	testCfgPath := "services.slam.attributes.fake"
+	logger := golog.NewTestLogger(t)
+
+	t.Run("Empty config", func(t *testing.T) {
+		model := resource.DefaultModelFamily.WithModel("test")
+		cfgService := resource.Config{Name: "test", API: slam.API, Model: model}
+		_, err := newConfig(cfgService)
+		test.That(t, err, test.ShouldBeError, newError("error validating \"services.slam.attributes.fake\": \"sensors\" must not be empty"))
+	})
+
+	testValidateTesthelper(t, logger, testCfgPath, false)
+	testValidateTesthelper(t, logger, testCfgPath, true)
```

Comment:
[q] so `testValidateHelper` goes through all tests, both with and without the imu feature flag?

Reply:
`testValidateHelper` goes through all of the tests although it looks like there were no existing tests tests where imuIntegrationEnabled was set to true